src/interfaceR.cpp

```diff
@@ -216,27 +218,35 @@ List reset_problem() {
 //[[Rcpp::export]]
 double c_eval(IntegerVector x) {
 	if(current_problem == NULL) {
-		return NULL;
+	  Rcout << "Error! No function selected.\n";
+		return -DBL_MAX;
 	}
 
 	double *result = IOHprofiler_allocate_vector(1);
-	
+
 	if(x.size() != current_problem->number_of_variables) {
 		Rcout << "Error! Dimension of input vector is incorrect.\n";
 	 	return -DBL_MAX;
 	}
 
-	int N = x.size();
-	int* x_ptr = (int*) malloc(sizeof(int) * N);
-	for (int i = 0; i < N; ++i) {
-		x_ptr[i] = x[i];
-	}
+	// int N = x.size();
+	// int* x_ptr = (int*) malloc(sizeof(int) * N);
+	// for (int i = 0; i < N; ++i) {
+	// 	x_ptr[i] = x[i];
+	// }
 
-	IOHprofiler_evaluate_function(current_problem, x_ptr, result);
-	free(x_ptr);
+	IOHprofiler_evaluate_function(current_problem, INTEGER(x), result);
```

Comment:
INTEGER(x) will convert a IntergerVector type to int* ? I am not aware of this.

Reply:
Furong says this is indeed the case, and it seems to work correctly

Reply:
Sure